### PR TITLE
docs: organize docs directory

### PR DESCRIPTION
**Pull request by mattrosno**
This PR cleans up the /docs directory and adds a README to help users find what they're looking for. Follow-on PRs will include migrating wiki pages to the /docs folder, as well as documenting project process.

Demo

* https://github.com/carbon-design-system/carbon/tree/9982391b52e9090faffa5e9cfcd5ba28f49913cb/docs

Changed

* `/docs` directory structure

New

* `/docs/README.md`

### CODE REVIEW

**Comment by mattrosno** [comment]
@carbon-design-system/developers any opinion on me moving wiki pages into our `/docs` directory? After doing so, either disable the wiki or have the one wiki page point to our repo docs? Right now our wiki content is pretty outdated and stale.

Advantage of using `/docs` instead of wiki is version control with context (via PRs) on why something was documented, we can run tooling (linting, formatting, table of contents generation) to keep markdown clean, etc.

Also, I'm adding structure to our docs in the form of a readme: https://github.com/carbon-design-system/carbon/tree/58fbf0bdce0d93a6599b5a389e2b19af29a80f43/docs

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Carbon documentation
+
+Welcome to the Carbon documentation. Here you'll find information on project
```
Comment (connor-leech) [inline docs/README.md]:
```suggestion
This is the home for all process, contribution, and version-related documentation. These docs include information on project
```

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+management, contribution, migration, lessons learned along the way, etc.
```
Comment (connor-leech) [inline docs/README.md]:
```suggestion
management, contribution, migration, reflections, and process insights.

If you think something is missing, we encourage [contributions](#how-to-contribute).
```

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+## Project management
+
+- [Endgame](/project-management/endgame.md)
+- [Versioning](/project-management/versioning.md)
+- [Support](/project-management/support.md)
+
+## Contributing
```
Comment (connor-leech) [inline docs/README.md:13]:
```suggestion
## How to contribute
```
These headers should reflect those on our site

File: docs/README.md (hunk 1 above)
Comment (mattrosno) [reply docs/README.md]: @connor-leech Why call out "not related to tooling"?

Comment (connor-leech) [reply docs/README.md]: Because this isn't an exhaustive repository of our documentation, it's primarily docs that don't live on our website.

Comment (mattrosno) [reply docs/README.md]: Do we have "tooling" documentation elsewhere?

Comment (connor-leech) [reply docs/README.md]: Our website? How to install Sketch, how to use our grid, icons, etc. Component usage guidance. 

If you think there's a better word than tooling I'd prefer it, as I typically don't like using that word.

Comment (mattrosno) [reply docs/README.md]: The /docs folder is for process, contribution, and version-related documentation. The website is usage guidelines and resources list.

Comment (connor-leech) [reply docs/README.md]: Brilliant, just updated 👍

**Review by connor-leech — APPROVED** [review]
Saweet

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -13,7 +13,7 @@ to the Carbon community.
 
 We value all of our community members, and thus want to foster a positive
 contributing environment. Please take a look at our
-[code of conduct](https://github.com/carbon-design-system/carbon/blob/master/docs/developer-handbook.md#carbon-code-of-conduct)
+[code of conduct](https://github.com/carbon-design-system/carbon/blob/master/docs/contributing/developer-handbook.md#carbon-code-of-conduct)
```
Comment (emyarod) [inline .github/CONTRIBUTING.md]: should we make these relative paths as well so they're viewable on branches?

**Review by emyarod — COMMENTED** [review]
looks like we also need to format with `prettier` to pass CI

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+## Project management
+
+- [Endgame](/project-management/endgame.md)
+- [Versioning](/project-management/versioning.md)
+- [Support](/project-management/support.md)
+
+## How to contribute
+
+- [Developer handbook](/contributing/developer-handbook.md)
```
Comment (joshblack) [inline docs/README.md:15]: Could you talk through some of the groupings? Not sure I understand all of them.

For example, Windows HCM is more related to support and definition and done over contributing. IBM Plex is more of a guide for using IBM Plex versus something folks do when working with the project

**Comment by joshblack** [comment]
bump if you have the time for above @mattrosno 👍

**Comment by mattrosno** [comment]
> Could you talk through some of the groupings? Not sure I understand all of them.

@joshblack I updated the explanation and hierarchy. Does this help?

https://github.com/carbon-design-system/carbon/tree/dc381215f30e651285f421d274de13d3f4ebcfa0/docs

**Comment by mattrosno** [comment]
@joshblack effectively, 1) here's how we work so you know what to expect and can plan accordingly. 2) here's how you can contribute. 3) here are guides for using Carbon (versioned technical guides not surfaced in the website). 4) here are miscellaneous guides for the core team.

**Pull request closed without merging**